testsuite/forge/src/backend/k8s/swarm.rs

```diff
@@ -165,11 +166,22 @@ impl Swarm for K8sSwarm {
             .get(version)
             .cloned()
             .ok_or_else(|| anyhow!("Invalid version: {:?}", version))?;
-        set_validator_image_tag(
+        set_stateful_set_image_tag(
             validator.stateful_set_name().to_string(),
-            version,
+            // the container name for the validator in its StatefulSet is "validator"
+            "validator".to_string(),
+            // extract the image tag from the "version"
+            version.to_string(),
             self.kube_namespace.clone(),
         )
+        .await?;
+
+        // If port-forward is enabled, we need to ensure that the validator is fully spun back up before attempting
+        // to port-forward it
+        if validator.port_forward_enabled {
```

Comment:
Don't we need to wait for validator to fully spun back up even if port forward is not enabled?

Reply:
That's true. I guess this is an optimization for the batch_upgrade. let me clean it up a bit